src/Microsoft.AspNetCore.AzureAppServices.HostingStartup/AzureKeyVaultHostingStartup.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Azure.KeyVault;
+using Microsoft.Azure.Services.AppAuthentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.AzureKeyVault;
+using Microsoft.Extensions.DependencyInjection;
+
+[assembly: HostingStartup(typeof(Microsoft.AspNetCore.AzureKeyVault.HostingStartup.AzureKeyVaultHostingStartup))]
+
+namespace Microsoft.AspNetCore.AzureKeyVault.HostingStartup
+{
+    /// <summary>
+    /// A dynamic KeyVault lightup experience
+    /// </summary>
+    public class AzureKeyVaultHostingStartup : IHostingStartup
+    {
+        private const string HostingStartupName = "KeyVault";
+        private const string ConfigurationFeatureName = "Configuration";
+        private const string ConfigurationVaultName = "ConfigurationVault";
+        private const string DataProtectionFeatureName = "DataProtection";
+        private const string DataProtectionKeyName = "DataProtectionKey";
+
+        /// <inheritdoc />
+        public void Configure(IWebHostBuilder builder)
+        {
+            var azureServiceTokenProvider = new AzureServiceTokenProvider();
+            var authenticationCallback = new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback);
+            var keyVaultClient = new KeyVaultClient(authenticationCallback);
+
+            var baseConfiguration = HostingStartupConfigurationExtensions.GetBaseConfiguration();
+
+            builder.ConfigureServices((context, collection) =>
+            {
+                var configuration = new ConfigurationBuilder()
+                    .AddConfiguration(baseConfiguration)
+                    .AddConfiguration(context.Configuration)
+                    .Build();
+
+                if (configuration.IsEnabled(HostingStartupName, DataProtectionFeatureName) &&
+                    configuration.TryGetOption(HostingStartupName, DataProtectionKeyName, out var protectionKey))
+                {
+                    AddDataProtection(collection, keyVaultClient, protectionKey);
+                }
+            });
+
+            if (baseConfiguration.IsEnabled(HostingStartupName, ConfigurationFeatureName) &&
```

Comment:
Why do you need two keys for this if you're always going to check both? Why not enable based only on the ConfigurationVaultName? Same for DataProtectionKeyName.

Reply:
Just to keep a same pattern for enabling features. Some of them wouldn't have parameters and I'd like to have a common way for enabling\disabling.

Reply:
Can the key name say "ConfigurationEnabled" at least?